### PR TITLE
use registry for defining viewer choices

### PR DESCRIPTION
**Pull request by kecnry**
This PR swaps out the manual definitions of individual viewers with iterating through the glue registry.  Note that:
* the viewers are only visible in the registry after they are actually imported, so histogram, image, and scatter are imported here.
* there is no current way from the registry to lookup the glue method for creating the viewer, so that mapping is manually created here along with a "manual" fallback which will not set all attributes correctly.

### CODE REVIEW

**Review by iisakkirotko — APPROVED** [review]
Hey @kecnry!

The PR looks good. If you can rebase it onto master, I'll merge it. Let me know if you'd rather I rebase it!

File: glue_solara/app.py [hunk 1]
```diff
@@ -86,17 +98,15 @@ def GlueApp(app: gj.JupyterApplication):
...
-            app.histogram1d(data=data, show=False)
-        elif type == "Scatter":
-            app.scatter2d(data=data, show=False)
-        elif type == "2D Image":
-            try:
-                app.imshow(data=data, show=False)
-            except ValueError as error:
-                error_message.set(str(error))
-                show_error.set(True)
-                return
```
Comment (kecnry) [inline glue_solara/app.py:99]: if this is still useful, the entire `getattr(app, ...)` line could be wrapped in this to keep it for all viewers.

**Review by kecnry — COMMENTED** [review]
Hi @iisakkirotko!  I rebased locally but don't seem to have permissions to force push.  The conflicts were quite straightforward though (mainly updating to keep the new grid layout changes from #11) if you'd rather just rebase on your end.

**Comment by iisakkirotko** [comment]
Thanks a lot!

**Pull request merged**